tensorboard/summary/writer/record_writer_test.py

```diff
@@ -62,6 +63,21 @@ def test_record_writer_roundtrip(self):
       r.GetNext()
       self.assertEqual(r.record(), bytes_to_write)
 
+  def test_record_immediate_read(self):
+    filename = os.path.join(self.get_temp_dir(), "record_immediate_read")
+    bytes_to_write = b"hello world"
```

Comment:
I think this would be a better test if we use different bytestrings each time (for example, a stringified counter value).  Otherwise, as is, this test would pass if we were actually returning the first record over and over.